go/vt/discovery/healthcheck.go

```diff
@@ -277,17 +277,37 @@ func (hc *HealthCheckImpl) servingConnStats() map[string]int64 {
 	return res
 }
 
+// finalizeConn closes the health checking connection and sends the final
+// notification about the tablet to downstream.
+func (hc *HealthCheckImpl) finalizeConn(hcc *healthCheckConn) {
+	hcc.mu.Lock()
+	if hcc.conn != nil {
+		hcc.conn.Close()
+		hcc.conn = nil
+	}
+	hcc.tabletStats.Up = false
+	hcc.tabletStats.Serving = false
+	hcc.tabletStats.LastError = hcc.ctx.Err()
```

Comment:
AFAICS, checkConn() has "return" only in two places, and both of them are when ctx.Done() is closed. Am I missing some other way the function can return?